core/src/main/java/org/bitcoinj/store/FlatDB.java

```diff
@@ -230,7 +230,7 @@ ReadResult read(Type object, boolean fDryRun) {
                 }
 
                 // de-serialize file header (network specific magic number) and ..
-                pchMsgTmp = (int)Utils.readUint32(vchData, magicMessage.length());
+                pchMsgTmp = Utils.readUint32(vchData, magicMessage.length());
```

Comment:
It turns out that casting to `int` and storing the value in a long will sign extend the value and make this fail:

`pchMsgTmp != context.getParams().getPacketMagic()`